addons/stock_dropshipping/models/stock.py

```diff
@@ -38,3 +38,6 @@ class StockPicking(models.Model):
     def _compute_is_dropship(self):
         for picking in self:
             picking.is_dropship = picking.location_dest_id.usage == 'customer' and picking.location_id.usage == 'supplier'
+
+    def show_delivery_address(self):
```

Comment:
You should perhaps rename this method ? It looks like, if it returns `True`, it is enough to display the delivery address. But, actually, it will need more conditions (`o.move_ids_without_package and o.move_ids_without_package[0].partner_id`)

Reply:
Or you take the whole condition ? It might be an alternative, I didn't check

Reply:
Something like
```diff
diff --git a/addons/stock/models/stock_picking.py b/addons/stock/models/stock_picking.py
index 94cff88ddb1..0bfa1df6ac9 100644
--- a/addons/stock/models/stock_picking.py
+++ b/addons/stock/models/stock_picking.py
@@ -779,6 +779,14 @@ class Picking(models.Model):
         self.write({'printed': True})
         return self.env.ref('stock.action_report_picking').report_action(self)
 
+    def should_print_delivery_address(self):
+        self.ensure_one()
+        return self.move_ids_without_package and self.move_ids_without_package[0].partner_id and self._is_to_external_location()
+
+    def _is_to_external_location(self):
+        self.ensure_one()
+        return self.picking_type_code == 'outgoing'
+
     def action_confirm(self):
         self._check_company()
         self.mapped('package_level_ids').filtered(lambda pl: pl.state == 'draft' and not pl.move_ids)._generate_moves()
diff --git a/addons/stock/report/report_deliveryslip.xml b/addons/stock/report/report_deliveryslip.xml
index eaff5798b8e..dcb15ccf520 100644
--- a/addons/stock/report/report_deliveryslip.xml
+++ b/addons/stock/report/report_deliveryslip.xml
@@ -8,12 +8,12 @@
 
                 <t t-set="address">
                     <div name="div_outgoing_address">
-                        <div t-if="o.move_ids_without_package and o.move_ids_without_package[0].partner_id and o.move_ids_without_package[0].partner_id != partner">
+                        <div t-if="o.should_print_delivery_address()">
                             <span><strong>Delivery Address:</strong></span>
                             <div t-field="o.move_ids_without_package[0].partner_id"
                                 t-options='{"widget": "contact", "fields": ["address", "name", "phone"], "no_marker": True, "phone_icons": True}'/>
                         </div>
-                        <div t-if="o.picking_type_id.code != 'internal' and (not o.move_ids_without_package or not o.move_ids_without_package[0].partner_id or o.move_ids_without_package[0].partner_id == partner) and o.picking_type_id.warehouse_id.partner_id">
+                        <div t-if="o.picking_type_id.code != 'internal' and not o.should_print_delivery_address() and o.picking_type_id.warehouse_id.partner_id">
                             <span><strong>Warehouse Address:</strong></span>
                             <div t-field="o.picking_type_id.warehouse_id.partner_id"
                                 t-options='{"widget": "contact", "fields": ["address", "name", "phone"], "no_marker": True, "phone_icons": True}'/>
diff --git a/addons/stock_dropshipping/models/stock.py b/addons/stock_dropshipping/models/stock.py
index 4a6f9bf5bd6..1d3a309562e 100644
--- a/addons/stock_dropshipping/models/stock.py
+++ b/addons/stock_dropshipping/models/stock.py
@@ -38,3 +38,6 @@ class StockPicking(models.Model):
     def _compute_is_dropship(self):
         for picking in self:
             picking.is_dropship = picking.location_dest_id.usage == 'customer' and picking.location_id.usage == 'supplier'
+
+    def _is_to_external_location(self):
+        return super()._is_to_external_location() or self.is_dropship
```

Reply:
I wonder if we could even use `t-elif`